src/azure-cli-core/azure/cli/core/tests/test_profile.py

```diff
@@ -470,7 +470,7 @@ def test_credscache_add_new_sp_creds(self, _, mock_open_for_write, mock_read_fil
             test_sp2['servicePrincipalTenant'])
 
         # assert
-        token_entries = [entry for _, entry in creds_cache.adal_token_cache.read_items()]
+        token_entries = [e for _, e in creds_cache.adal_token_cache.read_items()]  # noqa: F812
```

Comment:
F812: list comprehension redefines name from line N
http://flake8.pycqa.org/en/latest/user/error-codes.html

This is a rare case. Can't solve it completely because of the mock decorators.

Unfortunately, the flake8 doesn't an approach like pylint to disable error with a readable name. I think this is a very rare case.